Nov-28 15:59:30.232 [main] DEBUG nextflow.cli.Launcher - $> nextflow run main.nf
Nov-28 15:59:30.290 [main] INFO  nextflow.cli.CmdRun - N E X T F L O W  ~  version 23.10.0
Nov-28 15:59:30.304 [main] DEBUG nextflow.plugin.PluginsFacade - Setting up plugin manager > mode=prod; embedded=false; plugins-dir=/Users/lazarch2/.nextflow/plugins; core-plugins: nf-amazon@2.1.4,nf-azure@1.3.2,nf-cloudcache@0.3.0,nf-codecommit@0.1.5,nf-console@1.0.6,nf-ga4gh@1.1.0,nf-google@1.8.3,nf-tower@1.6.3,nf-wave@1.0.0
Nov-28 15:59:30.312 [main] INFO  o.pf4j.DefaultPluginStatusProvider - Enabled plugins: []
Nov-28 15:59:30.312 [main] INFO  o.pf4j.DefaultPluginStatusProvider - Disabled plugins: []
Nov-28 15:59:30.315 [main] INFO  org.pf4j.DefaultPluginManager - PF4J version 3.4.1 in 'deployment' mode
Nov-28 15:59:30.323 [main] INFO  org.pf4j.AbstractPluginManager - No plugins
Nov-28 15:59:30.334 [main] DEBUG nextflow.config.ConfigBuilder - Found config local: /Users/lazarch2/Desktop/wave/nextflow.config
Nov-28 15:59:30.335 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /Users/lazarch2/Desktop/wave/nextflow.config
Nov-28 15:59:30.348 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `standard`
Nov-28 15:59:30.671 [main] DEBUG nextflow.cli.CmdRun - Applied DSL=2 from script declararion
Nov-28 15:59:30.684 [main] INFO  nextflow.cli.CmdRun - Launching `main.nf` [furious_cajal] DSL2 - revision: a4d7b67a50
Nov-28 15:59:30.684 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins default=[]
Nov-28 15:59:30.684 [main] DEBUG nextflow.plugin.PluginsFacade - Plugins resolved requirement=[]
Nov-28 15:59:30.690 [main] DEBUG n.secret.LocalSecretsProvider - Secrets store: /Users/lazarch2/.nextflow/secrets/store.json
Nov-28 15:59:30.692 [main] DEBUG nextflow.secret.SecretsLoader - Discovered secrets providers: [nextflow.secret.LocalSecretsProvider@50448409] - activable => nextflow.secret.LocalSecretsProvider@50448409
Nov-28 15:59:30.732 [main] DEBUG nextflow.Session - Session UUID: 8f25be4c-3b00-4924-ae90-9423d3f743cd
Nov-28 15:59:30.732 [main] DEBUG nextflow.Session - Run name: furious_cajal
Nov-28 15:59:30.733 [main] DEBUG nextflow.Session - Executor pool size: 10
Nov-28 15:59:30.737 [main] DEBUG nextflow.file.FilePorter - File porter settings maxRetries=3; maxTransfers=50; pollTimeout=null
Nov-28 15:59:30.740 [main] DEBUG nextflow.util.ThreadPoolBuilder - Creating thread pool 'FileTransfer' minSize=10; maxSize=30; workQueue=LinkedBlockingQueue[10000]; allowCoreThreadTimeout=false
Nov-28 15:59:30.784 [main] DEBUG nextflow.cli.CmdRun - 
  Version: 23.10.0 build 5889
  Created: 15-10-2023 15:07 UTC (11:07 EDT)
  System: Mac OS X 13.6.1
  Runtime: Groovy 3.0.19 on OpenJDK 64-Bit Server VM 11.0.13+7-b1751.21
  Encoding: UTF-8 (UTF-8)
  Process: 70271@X32TNJL9CH [192.168.108.215]
  CPUs: 10 - Mem: 32 GB (2.9 GB) - Swap: 0 (0)
Nov-28 15:59:30.796 [main] DEBUG nextflow.Session - Work-dir: /Users/lazarch2/Desktop/wave/work [Mac OS X]
Nov-28 15:59:30.808 [main] DEBUG nextflow.executor.ExecutorFactory - Extension executors providers=[]
Nov-28 15:59:30.815 [main] DEBUG nextflow.Session - Observer factory: DefaultObserverFactory
Nov-28 15:59:30.826 [main] DEBUG nextflow.cache.CacheFactory - Using Nextflow cache factory: nextflow.cache.DefaultCacheFactory
Nov-28 15:59:30.834 [main] DEBUG nextflow.util.CustomThreadPool - Creating default thread pool > poolSize: 11; maxThreads: 1000
Nov-28 15:59:30.897 [main] DEBUG nextflow.Session - Session start
Nov-28 15:59:30.998 [main] DEBUG nextflow.script.ScriptRunner - > Launching execution
Nov-28 15:59:31.039 [main] DEBUG nextflow.executor.ExecutorFactory - << taskConfig executor: null
Nov-28 15:59:31.039 [main] DEBUG nextflow.executor.ExecutorFactory - >> processorType: 'local'
Nov-28 15:59:31.043 [main] DEBUG nextflow.executor.Executor - [warm up] executor > local
Nov-28 15:59:31.046 [main] DEBUG n.processor.LocalPollingMonitor - Creating local task monitor for executor 'local' > cpus=10; memory=32 GB; capacity=10; pollInterval=100ms; dumpInterval=5m
Nov-28 15:59:31.047 [main] DEBUG n.processor.TaskPollingMonitor - >>> barrier register (monitor: local)
Nov-28 15:59:31.104 [main] DEBUG nextflow.Session - Workflow process names [dsl2]: Cowpy
Nov-28 15:59:31.104 [main] DEBUG nextflow.Session - Igniting dataflow network (1)
Nov-28 15:59:31.104 [main] DEBUG nextflow.processor.TaskProcessor - Starting process > Cowpy
Nov-28 15:59:31.105 [main] DEBUG nextflow.script.ScriptRunner - Parsed script files:
  Script_94c0f0f5e2d20fa9: /Users/lazarch2/Desktop/wave/main.nf
Nov-28 15:59:31.105 [main] DEBUG nextflow.script.ScriptRunner - > Awaiting termination 
Nov-28 15:59:31.106 [main] DEBUG nextflow.Session - Session await
Nov-28 15:59:31.191 [Task submitter] DEBUG n.executor.local.LocalTaskHandler - Launch cmd line: /bin/bash -ue .command.run
Nov-28 15:59:31.193 [Task submitter] INFO  nextflow.Session - [1f/1a01b6] Submitted process > Cowpy
Nov-28 15:59:31.474 [Task monitor] DEBUG n.processor.TaskPollingMonitor - Task completed > TaskHandler[id: 1; name: Cowpy; status: COMPLETED; exit: 127; error: -; workDir: /Users/lazarch2/Desktop/wave/work/1f/1a01b632dde50036d45c00a6b4412e]
Nov-28 15:59:31.478 [Task monitor] DEBUG nextflow.processor.TaskProcessor - Handling unexpected condition for
  task: name=Cowpy; work-dir=/Users/lazarch2/Desktop/wave/work/1f/1a01b632dde50036d45c00a6b4412e
  error [nextflow.exception.ProcessFailedException]: Process `Cowpy` terminated with an error exit status (127)
Nov-28 15:59:31.487 [Task monitor] ERROR nextflow.processor.TaskProcessor - Error executing process > 'Cowpy'

Caused by:
  Process `Cowpy` terminated with an error exit status (127)

Command executed:

  cowpy.py

Command exit status:
  127

Command output:
  (empty)

Command error:
  env: python: No such file or directory

Work dir:
  /Users/lazarch2/Desktop/wave/work/1f/1a01b632dde50036d45c00a6b4412e

Tip: you can replicate the issue by changing to the process work dir and entering the command `bash .command.run`
Nov-28 15:59:31.491 [main] DEBUG nextflow.Session - Session await > all processes finished
Nov-28 15:59:31.491 [Task monitor] DEBUG nextflow.Session - Session aborted -- Cause: Process `Cowpy` terminated with an error exit status (127)
Nov-28 15:59:31.502 [main] DEBUG nextflow.Session - Session await > all barriers passed
Nov-28 15:59:31.502 [Task monitor] DEBUG n.processor.TaskPollingMonitor - <<< barrier arrives (monitor: local) - terminating tasks monitor poll loop
Nov-28 15:59:31.505 [main] DEBUG n.trace.WorkflowStatsObserver - Workflow completed > WorkflowStats[succeededCount=0; failedCount=1; ignoredCount=0; cachedCount=0; pendingCount=0; submittedCount=0; runningCount=0; retriesCount=0; abortedCount=0; succeedDuration=0ms; failedDuration=209ms; cachedDuration=0ms;loadCpus=0; loadMemory=0; peakRunning=1; peakCpus=1; peakMemory=0; ]
Nov-28 15:59:31.702 [main] DEBUG nextflow.cache.CacheDB - Closing CacheDB done
Nov-28 15:59:31.733 [main] DEBUG nextflow.script.ScriptRunner - > Execution complete -- Goodbye
